addons/web_editor/models/ir_attachment.py

```diff
@@ -6,60 +6,73 @@
 
+    @api.depends('mimetype')
+    def _compute_filetype(self):
+        for attachment in self:
+            attachment.filetype = 'image' if attachment.mimetype in SUPPORTED_VIDEO_MIMETYPES else 'image'
+
     def _compute_local_url(self):
         for attachment in self:
             if attachment.url:
                 attachment.local_url = attachment.url
             else:
-                attachment.local_url = '/web/image/%s?unique=%s' % (attachment.id, attachment.checksum)
+                attachment.local_url = f'/web/{attachment.filetype}/{attachment.id}?unique={attachment.checksum}'
 
     @api.depends('mimetype', 'url', 'name')
-    def _compute_image_src(self):
+    def _compute_src(self):
         for attachment in self:
             # Only add a src for supported images
-            if attachment.mimetype not in SUPPORTED_IMAGE_MIMETYPES:
-                attachment.image_src = False
+            if attachment.mimetype not in SUPPORTED_IMAGE_MIMETYPES + SUPPORTED_VIDEO_MIMETYPES:
+                attachment.src = False
                 continue
 
             if attachment.type == 'url':
-                attachment.image_src = attachment.url
+                attachment.src = attachment.url
             else:
                 # Adding unique in URLs for cache-control
                 unique = attachment.checksum[:8]
                 if attachment.url:
                     # For attachments-by-url, unique is used as a cachebuster. They
                     # currently do not leverage max-age headers.
                     separator = '&' if '?' in attachment.url else '?'
-                    attachment.image_src = '%s%sunique=%s' % (attachment.url, separator, unique)
+                    attachment.src = f'{attachment.url}{separator}unique={unique}'
                 else:
                     name = url_quote(attachment.name)
-                    attachment.image_src = '/web/image/%s-%s/%s' % (attachment.id, unique, name)
+                    attachment.src = f'/web/{attachment.filetype}/{attachment.id}-{unique}/{name}'
 
     @api.depends('datas')
-    def _compute_image_size(self):
+    def _compute_size(self):
         for attachment in self:
-            try:
-                image = tools.base64_to_image(attachment.datas)
-                attachment.image_width = image.width
-                attachment.image_height = image.height
-            except Exception:
-                attachment.image_width = 0
-                attachment.image_height = 0
+            if attachment.filetype == "image":
+                try:
+                    image = tools.base64_to_image(attachment.datas)
+                    attachment.width = image.width
+                    attachment.height = image.height
+                except Exception:
+                    attachment.width = 0
+                    attachment.height = 0
+            else:
+                attachment.height = 0
+                attachment.width = 0
```

Comment:
They should be at some point for the videos, if we want to implement custom size rendering, maybe compression, etc. The point is that if we're extending ir_attachment to videos then we should be prepared for more than just simple upload and processing (like cropping, trimming, etc.) would require the size eventually. It also makes sense for them to be generic instead of specific, if we ever extend the ir.attachment to cover more file formats and types (pdf rendering maybe?)